tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -108,15 +108,19 @@ sub run {
     my $self = shift;
     my $timeout = get_var('MAX_TEST_TIME', '36000') + 10;
     my $upload_log_name = 'guest-upgrade-logs';
-    script_run("echo \"Debug info: max_test_time is $timeout\"");
+    # Add new option die_on_timeout=>0 to adapt API script_run() new behavior change
+    # Refer to poo#105720 for more details
+    my %args = ();
+    $args{die_on_timeout} = 0;
+    script_run("echo \"Debug info: max_test_time is $timeout\"", %args);
     # Modify source configuration file sources.* of virtauto-data pkg on host
     # to use openqa daily build installer repo and module repo for guests,
     # and it will be copied into guests to be used during guest upgrade test
     repl_module_in_sourcefile();
 
     $self->{'package_name'} = 'Guest Upgrade Test';
     if (is_s390x) {
-        script_run "[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /var/lib/libvirt/images/prj4_guest_upgrade ] && rm -r /var/lib/libvirt/images/prj4_guest_upgrade /tmp/full_guest_upgrade_test-* /tmp/kill_zypper_procs-* /tmp/update-guest-*";
+        script_run "[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /var/lib/libvirt/images/prj4_guest_upgrade ] && rm -r /var/lib/libvirt/images/prj4_guest_upgrade /tmp/full_guest_upgrade_test-* /tmp/kill_zypper_procs-* /tmp/update-guest-*", %args;
```

Comment:
Why not 'do not die' on timeout. I thin this should die on timeout.

Reply:
refer to current osd s390x test log: 
```
[2022-01-29T12:02:38.361088+01:00] [debug] <<< testapi::script_run(cmd="sed -i 's/#dnssec-validation auto;/dnssec-validation no;/g' /etc/named.conf", die_on_timeout=-1, output="", quiet=undef, timeout=30)
[2022-01-29T12:02:38.361285+01:00] [debug] tests/virt_autotest/update_package.pm:64 called testapi::script_run
[2022-01-29T12:02:38.361443+01:00] [debug] <<< testapi::type_string(string="sed -i 's/#dnssec-validation auto;/dnssec-validation no;/g' /etc/named.conf", max_interval=250, wait_screen_changes=0, wait_still_screen=0, timeout=30, similarity_level=47)
[2022-01-29T12:02:40.918178+01:00] [debug] tests/virt_autotest/update_package.pm:64 called testapi::script_run
[2022-01-29T12:02:40.918415+01:00] [debug] <<< testapi::type_string(string="; echo q85AM-\$?- > /dev/ttysclp0\n", max_interval=250, wait_screen_changes=0, wait_still_screen=0, timeout=30, similarity_level=47)
[2022-01-29T12:02:42.121864+01:00] [debug] tests/virt_autotest/update_package.pm:64 called testapi::script_run
[2022-01-29T12:02:42.122290+01:00] [debug] <<< testapi::wait_serial(buffer_size=undef, quiet=undef, record_output=undef, timeout=30, expect_not_found=0, no_regex=0, regexp=qr/q85AM-\d+-/u)
[2022-01-29T12:03:13.231538+01:00] [debug] >>> testapi::wait_serial: (?^u:q85AM-\d+-): fail
[2022-01-29T12:03:13.231688+01:00] [warn] !!! testapi::script_run: DEPRECATED call of script_run() in tests/virt_autotest/update_package.pm:64 add `die_on_timeout => ?` to the call or set
                        $distri->{script_run_die_on_timeout} to avoid this
                        warning
```
due to the above warning due to use with the default setting die_on_timeout=-1.
Just try to fix up the above warning info and keep the old  behavior die_on_timeout => 0, as this PR purpose.